core/modules/user/components/UserShippingDetails.ts

```diff
@@ -111,18 +112,29 @@ export const UserShippingDetails = {
           })
         }
       }
+      updatedShippingDetails = pick(updatedShippingDetails, config.users.allowModification)
       this.exitSection(null, updatedShippingDetails)
     },
     exitSection (event, updatedShippingDetails) {
       this.$bus.$emit('myAccount-before-updateUser', updatedShippingDetails)
-      if (!updatedShippingDetails) {
-        this.shippingDetails = this.getShippingDetails()
-        this.useCompanyAddress = false
-        this.remainInEditMode = false
-      }
-      if (!this.remainInEditMode) {
-        this.isEdited = false
-      }
+      userHooks.afterUserProfileUpdated(async (event: any) => {
+        if (event.resultCode === 200) {
+          if (!updatedShippingDetails) {
+            this.shippingDetails = this.getShippingDetails()
+            this.useCompanyAddress = false
+            this.remainInEditMode = false
+          }
+          if (!this.remainInEditMode) {
+            this.isEdited = false
+          }
+        } else {
+          await this.$store.dispatch('notification/spawnNotification', {
```

Comment:
Same question as above with this `await`

Reply:
done